src/nvim/decoration.c

```diff
@@ -201,25 +232,25 @@ bool decor_redraw_start(buf_T *buf, int top_row, DecorState *state)
     }
 
 next_mark:
-    if (marktree_itr_node_done(state->itr)) {
-      marktree_itr_next(buf->b_marktree, state->itr);
-      break;
-    }
+    // TODO(lewis6991): Need to iter the full tree for signs.
+    // if (marktree_itr_node_done(state->itr)) {
+    //   marktree_itr_next(buf->b_marktree, state->itr);
+    //   break;
+    // }
```

Comment:
I don't fully understand why this code was needed, but it was causing the sign tests to fail. Is this change ok? Why did we stop iterating when reaching the end of a tree node?

Reply:
This is to find any range that starts or end within the current node. after that `decor_redraw_col()` is advancing the iterator as we keep rendering.

Reply:
So what would be the correct fix here?

Reply:
I've tried studying marktree.c and decoration.c but not many of the functions are documented.

From what I understand, we need to comment this out to force a full scan of the remaining marks in order to catch signs that begin above the top window line but end below the bottom window line. This is fairly expensive to what we are doing now but can be made more efficient when #13243 is done.

@bfredl is this correct analysis?

Reply:
@lewis6991 More or less. but for now we should just allow such signs to go missing rather scanning the entire tree, before #13243 is done (realistically, I'll could get it done for 0.8). This already works well enough for multi-line highlights, those work if they are small and don't overlap too many other marks (the common case), and if they are big we are already using multiple marks as a work around. Scanning the entire tree here defeats the purpose of having the btree data structure in the first place.